### ISSUE TITLE
Text Selection with a Cursor

### ISSUE THREAD

**Issue by dit7ya** [tridactyl/tridactyl#2287]
Tridactyl version: 1.17.1pre3754-99fdbe6
Firefox version: Mozilla Firefox 74.0
Operating system: linux
<!--

I know this is somewhat implemented in the new visual mode , but it would be nice to have more specificity in text selection that a cursor can offer. [SurfigKeys](https://github.com/brookhong/Surfingkeys) has an implementation of this.

**Comment by bovine3dom** [comment]
Do you just mean a visual representation of the cursor?

**Comment by dit7ya** [comment]
Apart from the visual mode selection that we have currently, here's what I would like to use.

There is a keybinding that shows hints on text objects and after choosing a hint takes me to the `text-selection-mode` which shows a cursor next to the text object. If I am not happy  with the location of it, I can use vim navigation keys to move it around. Once I am happy about the beginning of the selection, I press another keybinding which shows hints again and after choosing a hint, selects the text between these two places. Then it again shows a different cursors for controlling the end of the selection. 

All this will provide is a finer control over text selection. Hope I am clear. I would be happy if all this procedure can be reduced to even lesser number of keypresses, but without trading off the specificity.

**Comment by bovine3dom** [comment]
So, in essence, you want to be able to extend the selection manually in both directions? Would a bind that just swaps the cursor from the beginning of the selection to the end of it be good enough?

You may find that the current `=` bind which expands the selection is useful if you don't already know of it.

**Comment by dit7ya** [comment]
Yeah, swapping would be great (maybe a smoother experience than what I originally proposed, cannot be sure without trying it out). 

Suppose I want to select three words from a line from your comment above. If I understand correctly, this is what the workflow would be
- select the entire line with visual mode
- go back few words with `b`
- swap the cursor to beginning of the line with a keybinding
- go forward few words with `w`


Thanks for the `=` bind, didn't know it exists.

**Comment by lydell** [comment]
Here’s a function to reverse the selection, in case it’s useful:

https://github.com/lydell/LinkHints/blob/4e7ec695b15e1cda474ed4271b8c7c054aacf207/src/worker/Program.js#L1005-L1039

**Comment by bovine3dom** [comment]
`o` bind using lydell's function added in e749ed08fe9a284e86824d885a840c0c1fdb4433...426513327da00e316ea45cefc40cd7997f8e2153

It could be improved a lot, I think, by having the cursor be visible. I'll leave this issue open for that.

**Comment by dit7ya** [comment]
Nice! Just tried it. Having the cursor visible, or having some kind of marker in the mode-button would be great.

BTW, is it just me or incrementing selection in visual mode is buggy for everyone else? By buggy I mean it would randomly lose the selection.

**Comment by bovine3dom** [comment]
Yeah, it is buggy. I haven't looked into why yet.
